src/range_proof.rs

```diff
@@ -727,21 +743,16 @@ where
             // Compute useful challenge values
             let z_square = z * z;
             let e_square = e * e;
-            let mut y_nm = y;
-            let mut challenges_sq = Vec::with_capacity(challenges.len());
-            let mut challenges_sq_inv = Vec::with_capacity(challenges_inv.len());
-            for i in 0..rounds {
-                y_nm = y_nm * y_nm;
-                challenges_sq.push(challenges[i] * challenges[i]);
-                challenges_sq_inv.push(challenges_inv[i] * challenges_inv[i]);
-            }
+            let challenges_sq: Vec<Scalar> = challenges.iter().map(|c| c * c).collect();
+            let challenges_sq_inv: Vec<Scalar> = challenges_inv.iter().map(|c| c * c).collect();
```

Comment:
Is it possible for challenge*challenge to overflow?

Reply:
Nope. Ristretto scalar multiplication is always a well-defined operation to which overflow does not apply.